homeassistant/components/motioneye/sensor.py

```diff
@@ -0,0 +1,89 @@
+"""Sensor platform for motionEye."""
+from __future__ import annotations
+
+import logging
+from types import MappingProxyType
+from typing import Any
+
+from motioneye_client.client import MotionEyeClient
+from motioneye_client.const import KEY_ACTIONS, KEY_NAME
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import MotionEyeEntity, get_camera_from_cameras, listen_for_new_cameras
+from .const import CONF_CLIENT, CONF_COORDINATOR, DOMAIN, TYPE_MOTIONEYE_ACTION_SENSOR
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up motionEye from a config entry."""
+    entry_data = hass.data[DOMAIN][entry.entry_id]
+
+    @callback
+    def camera_add(camera: dict[str, Any]) -> None:
+        """Add a new motionEye camera."""
+        async_add_entities(
+            [
+                MotionEyeActionSensor(
+                    entry.entry_id,
+                    camera,
+                    entry_data[CONF_CLIENT],
+                    entry_data[CONF_COORDINATOR],
+                    entry.options,
+                )
+            ]
+        )
+
+    listen_for_new_cameras(hass, entry, camera_add)
+
+
+class MotionEyeActionSensor(MotionEyeEntity, SensorEntity):
+    """motionEye action sensor camera."""
+
+    def __init__(
+        self,
+        config_entry_id: str,
+        camera: dict[str, Any],
+        client: MotionEyeClient,
+        coordinator: DataUpdateCoordinator,
+        options: MappingProxyType[str, str],
+    ) -> None:
+        """Initialize an action sensor."""
+        super().__init__(
+            config_entry_id,
+            TYPE_MOTIONEYE_ACTION_SENSOR,
+            camera,
+            client,
+            coordinator,
+            options,
+            False,
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        camera_prepend = f"{self._camera[KEY_NAME]} " if self._camera else ""
```

Comment:
if self._camera is None then the MotionEyeActionSensor should just be unavailable

Reply:
`_camera` will only be None if the camera has been removed (camera is no longer in the downloaded configuration from motionEye). If this happens, the entities for that camera will be automatically removed, so making it unavailable is not necessary. However, there is a small window before the entity is removed in which this method may be called, and so dealing with the potential for _camera to be None is still likely necessary.

So whilst I think you're right (and in fact I even made the change you suggested, and couldn't figure out why the unittest didn't work!) -- I think it's not actually necessary to make changes given the above.

Reply:
Thought about this some more -- even though the entity is ~immediately removed, I think it's still better to follow your  advice and mark the entity unavailable.